ts/src/profiler.ts

```diff
@@ -120,6 +142,69 @@ async function profileBytes(p: perftools.profiles.IProfile): Promise<string> {
   return gzBuf.toString('base64');
 }
 
+/**
+ * Error constructed from http server response which indicates backoff.
+ */
+class BackoffResponseError extends Error {
+  constructor(response: http.ServerResponse, readonly backoffMillis: number) {
+    super(response.statusMessage);
+  }
+}
+
+/**
+ * @return true if error is a BackoffResponseError and false otherwise
+ */
+function isBackoffResponseError(err: Error): err is BackoffResponseError {
+  // tslint:disable-next-line: no-any
+  return typeof (err as any).backoffMillis === 'number';
+}
+
+/**
+ * Class which tracks how long to wait before the next retry and can be
+ * used to get this backoff.
+ */
+export class Retryer {
+  private nextBackoffMillis: number;
+  constructor(
+      readonly initialBackoffMillis: number, readonly maxBackoffMillis: number,
```

Comment:
maxBackoffMillis -> backoffCapMillis since that's the terminology you use now?

Reply:
Yes -- Done.